File: Assignment.py
import requests

url = f"https://samples.openweathermap.org/data/2.5/forecast/hourly?q=London,us&appid=b6907d289e10d714a6e88b30761fae22"

def get_weather(date):
    response = requests.get(url)
    data = response.json()
    for i in data["list"]:
        if i["dt_txt"].startswith(date):
            return i["main"]["temp"]

def get_wind_speed(date):
    response = requests.get(url)
    data = response.json()
    for i in data["list"]:
        if i["dt_txt"].startswith(date):
            return i["wind"]["speed"]


def get_pressure(date):
    response = requests.get(url)
    data = response.json()
    for i in data["list"]:
        if i["dt_txt"].startswith(date):
            return i["main"]["pressure"]

while True:
    print("\nselect 1 option from below:")
    print("1. Get weather")
    print("2. Get Wind Speed")
    print("3. Get Pressure")
    print("0. Exit")
    choice = input("Enter your choice: ")

    if choice == "1":
        date = input("Enter the date (YYYY-MM-DD): ")
        temp = get_weather(date)
        celsius = temp - 273.15
        print(f"The temperature on {date} is {temp}°K and {celsius:.2f}°C.")
    elif choice == "2":
        date = input("Enter the date (YYYY-MM-DD): ")
        wind_speed = get_wind_speed(date)
        print(f"The wind speed on {date} is {wind_speed} m/s.")
    elif choice == "3":
        date = input("Enter the date (YYYY-MM-DD): ")
        pressure = get_pressure(date)
        print(f"The pressure on {date} is {pressure} hPa.")
    elif choice == "0":
        print("Exiting the program.")
        break
    else:
        print("Select appropriate option")

File: Assignment.js
const axios = require("axios");

const url = "https://samples.openweathermap.org/data/2.5/forecast/hourly?q=London,us&appid=b6907d289e10d714a6e88b30761fae22";

function getWeather(date) {
  return axios.get(url)
    .then(response => {
      const data = response.data;
      for (const i of data.list) {
        if (i.dt_txt.startsWith(date)) {
          const temp = i.main.temp;
          const celsius = temp - 273.15;
          return {
            temperatureKelvin: temp,
            temperatureCelsius: celsius.toFixed(2)
          };
        }
      }
    });
}

function getWindSpeed(date) {
  return axios.get(url)
    .then(response => {
      const data = response.data;
      for (const i of data.list) {
        if (i.dt_txt.startsWith(date)) {
          return i.wind.speed;
        }
      }
    });
}

function getPressure(date) {
  return axios.get(url)
    .then(response => {
      const data = response.data;
      for (const i of data.list) {
        if (i.dt_txt.startsWith(date)) {
          return i.main.pressure;
        }
      }
    });
}

function main() {
  while (true) {
    console.log("\nselect 1 option from below:");
    console.log("1. Get weather");
    console.log("2. Get Wind Speed");
    console.log("3. Get Pressure");
    console.log("0. Exit");

    const readline = require("readline").createInterface({
      input: process.stdin,
      output: process.stdout
    });

    readline.question("Enter your choice: ", async (choice) => {
      if (choice === "1") {
        const date = await getInput("Enter the date (YYYY-MM-DD): ");
        const weatherData = await getWeather(date);
        console.log(`The temperature on ${date} is ${weatherData.temperatureKelvin}°K and ${weatherData.temperatureCelsius}°C.`);
      } else if (choice === "2") {
        const date = await getInput("Enter the date (YYYY-MM-DD): ");
        const windSpeed = await getWindSpeed(date);
        console.log(`The wind speed on ${date} is ${windSpeed} m/s.`);
      } else if (choice === "3") {
        const date = await getInput("Enter the date (YYYY-MM-DD): ");
        const pressure = await getPressure(date);
        console.log(`The pressure on ${date} is ${pressure} hPa.`);
      } else if (choice === "0") {
        console.log("Exiting the program.");
        process.exit();
      } else {
        console.log("Select appropriate option");
      }

      readline.close();
    });
  }
}

function getInput(prompt) {
  const readline = require("readline").createInterface({
    input: process.stdin,
    output: process.stdout
  });

  return new Promise((resolve) => {
    readline.question(prompt, (input) => {
      resolve(input);
      readline.close();
    });
  });
}

main();
